client/scripts/controllers/chain/cosmos/tokenAdapter.ts

```diff
@@ -0,0 +1,41 @@
+import Cosmos from 'controllers/chain/cosmos/main';
+
+import { NodeInfo, ITokenAdapter, ChainInfo } from 'models';
+import $ from 'jquery';
+import { IApp } from 'state';
+import BN from 'bn.js';
+
+export default class Token extends Cosmos implements ITokenAdapter {
```

Comment:
Ideally we should implement these features directly on Cosmos main rather than in a separate file, as all Cosmos communities can be treated "as" tokens for the sake of balance checking.